src/main/java/org/folio/list/services/CustomTenantService.java

```diff
@@ -1,37 +1,138 @@
 import org.folio.spring.service.PrepareSystemUserService;
 import org.folio.spring.service.TenantService;
 import org.folio.tenant.domain.dto.TenantAttributes;
+import org.jetbrains.annotations.NotNull;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Primary;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Service;
 
+import java.util.*;
+
 @Log4j2
 @Primary
 @Service
 public class CustomTenantService extends TenantService {
 
   protected final PrepareSystemUserService prepareSystemUserService;
+  private final ListRepository listRepository;
+  @Autowired
+  private ObjectMapper objectMapper;
 
   @Autowired
   public CustomTenantService(
     JdbcTemplate jdbcTemplate,
     FolioExecutionContext context,
     FolioSpringLiquibase folioSpringLiquibase,
-    PrepareSystemUserService prepareSystemUserService
+    PrepareSystemUserService prepareSystemUserService,
+    ListRepository listRepository
   ) {
     super(jdbcTemplate, context, folioSpringLiquibase);
     this.prepareSystemUserService = prepareSystemUserService;
+    this.listRepository = listRepository;
   }
 
   @Override
   protected void afterTenantUpdate(TenantAttributes tenantAttributes) {
     log.info("Initializing system user");
-    prepareSystemUserService.setupSystemUser();
+   prepareSystemUserService.setupSystemUser();
+    Collection<ListEntity> lists = (Collection<ListEntity>) listRepository.findAll();
+    for (ListEntity list : lists) {
+      System.out.println(list.getEntityTypeId());
+      if (list.getEntityTypeId().equals(UUID.fromString("0069cf6f-2833-46db-8a51-8934769b8289"))) {
+        list.setEntityTypeId(UUID.fromString("ddc93926-d15a-4a45-9d9c-93eadc3d9bbf"));
+        try {
+          updateFqlQueryForUsers(list);
+          updateFieldNamesForUsers(list);
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }
+      listRepository.save(list);
+    }
+  }
+
+  public void updateFqlQueryForUsers(ListEntity list) throws Exception {
+    Map<String, String> keyMappings = getStringStringMap();
+    String currentQuery = list.getFqlQuery();
+    Map<String, Object> map = objectMapper.readValue(currentQuery, new TypeReference<HashMap<String, Object>>() {});
+
+    Map<String, Object> result = new HashMap<>();
+
+    for (Map.Entry<String, Object> entry : map.entrySet()) {
+      String key = entry.getKey();
+      System.out.println("print key" +key);
+      if (keyMappings.containsKey(key)) {
+        String newKey = keyMappings.get(key);
+        result.put(newKey, entry.getValue());
+      } else {
+        result.put(key, entry.getValue());
+      }
+    }
+    String resultStr = objectMapper.writeValueAsString(result);
+    System.out.print("new migrated query" +resultStr);
+    list.setFqlQuery(resultStr);
+  }
+
+
+  public void updateFieldNamesForUsers(ListEntity list){
```

Comment:
Maybe instead of modifying the list in this method, make it return the updated fields, then update the list in afterTenantUpdate(). That way, this method can be free of side effects, making it a little easier to work with